### PR TITLE
AudioConfigSequence cleanup

### PR DESCRIPTION
**Pull request by DSDubit**
Changed sequences to return configs, not the raw AudioSource objects, since these should be handled internally by the audio system

### CODE REVIEW

File: AudioSystem/Editor/AudioConfigSequenceEditor.cs [hunk 1]
```diff
@@ -45,7 +45,7 @@ private void DrawEntry(SerializedProperty property)
 				}
 				EditorGUILayout.EndHorizontal();
 
-				var postClipDelay = property.FindPropertyRelative("postClipDelay");
+				var postClipDelay = property.FindPropertyRelative("preClipDelay");
```
Comment (Rollo-Liu) [inline AudioSystem/Editor/AudioConfigSequenceEditor.cs]: Need matching the name. `var preClipDelay`

File: AudioSystem/AudioSystemExtension.cs [hunk 2]
```diff
@@ -159,55 +159,66 @@ public static AudioSource Play(this AudioConfig audioConfig, float volume, Trans
...
 
 			if (index == 0 || delay <= 0f)
 			{
-				return playNextConfig();
+				return playThisConfig();
 			}
 			else
 			{
-				Timer.SetTimeout(delay, () => playNextConfig());
+				var timer = Timer.SetTimeout(delay, () => playThisConfig());
```
Comment (Rollo-Liu) [inline AudioSystem/AudioSystemExtension.cs]: I think you can pass in the function straight away?

like `var timer = Timer.SetTimeout(delay, playThisConfig);`

File: AudioSystem/Editor/AudioConfigSequenceEditor.cs (hunk 1 above)
Comment (DSDubit) [reply AudioSystem/Editor/AudioConfigSequenceEditor.cs]: Done

File: AudioSystem/AudioSystemExtension.cs (hunk 2 above)
Comment (DSDubit) [reply AudioSystem/AudioSystemExtension.cs]: Apparently not

**Pull request merged**